js/models/conversations.js

```diff
@@ -976,19 +980,39 @@
       if (!response) {
         return;
       }
-      const primaryConversation = ConversationController.get(
+
+      // Accept FRs from all the user's devices
+      const allDevices = await libloki.storage.getAllDevicePubKeysForPrimaryPubKey(
         this.getPrimaryDevicePubKey()
       );
-      // Should never happen
-      if (!primaryConversation) {
+
+      if (!allDevices.length) {
         return;
       }
-      const pending = await primaryConversation.getFriendRequests(
-        direction,
-        status
+
+      const allConversationsWithUser = allDevices.map(d =>
+        ConversationController.get(d)
       );
+
+      const pendingRequests = await allConversationsWithUser.reduce(
+        async (requestsP, conversation) => {
+          const requests = await requestsP;
+          const request = (await conversation.getFriendRequests(
+            direction,
+            status
+          ))[0];
+
+          return request
+            ? requests.concat({ conversation, request })
+            : requests;
+        },
+        []
+      );
```

Comment:
this part looks quite complicated. I am not even sure to understand what it does.

what about
would it be the same as this? 
```js
const promises = allConversationsWithUser.map(c => conversation.getFriendRequests(
            direction,
            status
          ))[0]);

let pendingRequests = await Promise.all(promises);
pendingRequests = pendingRequests.filter(p => !!p);
```

Reply:
Simplified